### PR TITLE
feat: OpenLLM

### PR DESCRIPTION
**Pull request by aarnphm**
🎉 OpenLLM 🤝 LangChain

[OpenLLM](https://github.com/bentoml/OpenLLM) is a new open platform for operating large language models(LLMs) in
production. Serve, deploy, and monitor any LLMs with ease.

OpenLLM lets developers and researchers to easily run inference with any
open-source LLMs, deploy to the cloud or on-premises, build powerful AI apps,
and fine tune your own LLM (coming soon...)

It currently supports ChatGLM, Dolly-v2, Flan-T5, Falcon, Starcoder, and more to
come. One can also easily start either a REST/gRPC server, which is powered by
BentoML.

Now that's out of the way, lets dive in!

The current depenedencies for this integration: `openllm`

This integrations brings a `OpenLLM` llms to LangChain, that can be used for
both running LLMs locally as well as interacting with a remote OpenLLM server.

To quickly start a local LLM, simply do the following:

```python
from langchain.llms import OpenLLM

llm = OpenLLM(model_name="dolly-v2", model_id='databricks/dolly-v2-7b', device_map='auto')

llm("What is the difference between a duck and a goose? And why there are so many Goose in Canada?")
```

`langchain.llms.OpenLLM`, as mentioned above also have the capabilities to
interact with remote OpenLLM Server. Given there is a running OpenLLM server at
http://44.23.123.1, you can do the following:

```python
from langchain.llms import OpenLLM

llm = OpenLLM(server_url='http://44.23.123.1:3000', server_type='grpc')

llm("What is the difference between a duck and a goose? And why there are so many Goose in Canada?")
```

Features soon to be open-sourced:

- OpenAI compatible API, allowing users to easily use LangChain's OpenAI llm.
- SSE support for OpenLLM server, allowing users to stream inference results.
- Last but not least, easily fine-tune your own LLMs with `LLM.tuning()`

Last but not least, I would love to hear feedback and response from the
community about the project, and feel free to reach out to me via Twitter
@aarnphm_. 

Feel free to join our [Discord](https://l.bentoml.com/join-openllm-discord) to get the latest updates and developments.

Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

#### Before submitting

I have added tests for this integration.

#### Who can review?

Tag maintainers/contributors who might be interested:

cc @hwchase17 @agola11

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
a notebook example would be great! thannks

File: docs/integrations/openllm.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Wrapper for Local Inference
+
+You can also use the OpenLLM wrapper to load LLM in current Python process for running inference.
+
+```python
+from langchain.llms import OpenLLM
+
+llm = OpenLLM("dolly-v2", model_id='databricks/dolly-v2-7b', device_map='auto')
```
Comment (aarnphm) [inline docs/integrations/openllm.md:53]:
```suggestion
llm = OpenLLM(model_name="dolly-v2", model_id='databricks/dolly-v2-7b', device_map='auto')
```

**Comment by hwchase17** [comment]
@dev2049 some conflicts with docs, gonna defer to you here

**Comment by aarnphm** [comment]
Hopefully I fixed the docs 😃

**Comment by aarnphm** [comment]
kindly cc @hwchase17 for another round

**Comment by dev2049** [comment]
merged and fixed some lint issues here if you want to merge this in https://github.com/hwchase17/langchain/compare/feat/openllm?expand=1

**Comment by aarnphm** [comment]
Hi @dev2049, Thanks for the help. I have updated accordingly

**Comment by aarnphm** [comment]
Hi @dev2049, I just want to confirm that you will take this from here?

**Comment by dev2049** [comment]
yep, am about to land to master in #6578! sorry thought something was wrong with poetry.lock but actually looks like it was all good 👍

**Comment by aarnphm** [comment]
Sounds good. Thanks again for all the help!

**Pull request merged**